### ISSUE TITLE
rfunc in StressModelBase should be able to be None

### ISSUE THREAD

**Issue by raoulcollenteur** [pastas/pastas#186]
Right now the StressModelBase always requires an rfunc. Some stress models have no response function (e.g., LinearTrend and FactorModel), so they internally provide One as the rfunc. This does not make much sense. rfunc should simply be able to be None.

This also leads to the following weird code in model.py that does nothing, (see method ml.get_response):

```
if not hasattr(self.stressmodels[name], "rfunc"):
    raise TypeError("Stressmodel {} has no rfunc".format(name))
```

It would be much better if we could simply check if a stress model has an rfunc registered, right now it always has a rfunc. 

Therefore I propose to change the default value of rfunc to None and remove the related arguments (cutoff, meanstress, up). The latter are not stored in StressModelBase anyway. In each stress model we then simply provide an instance of an rfunc to the `StressModelBase.__init__ `method (e.g., rfunc=ps.Gamma(up=up, meanstress=meanstress, cutoff=cutoff).

This would be an internal change only and remove some weird passing of unused arguments around. It would not effect any production code, and make StressModelBase ready for future stress models that do not use response functions.

**Comment by rubencalje** [comment]
I agree, changing this would be much cleaner.

**Comment by raoulcollenteur** [comment]
Closed as per #191
